### ISSUE TITLE
Remove a position which are greater then x kilometer from its previous position

### ISSUE THREAD

**Issue by pierrot10** [vitalidze/traccar-web#296]
Good day,
Somme time a position is more than 10km even more , from the previous and next position. It look like a "lost" position.
Would it possible to add filter which remove a position (from the event table)  if that position is more than x kilometer from its previous and the next position?

I would also suggest t remove all future position.
Occasionally some GPS tracking devices will send an event to the server with a date/time which is in the future, sometimes a long time into the future. This can be caused by marginal GPS converage.

Thank for you great job

**Comment by vitalidze** [comment]
You can filter positions on the "backend" side before they are posted to database. You can set up these filters in configuration file. See [this article](https://www.traccar.org/configuration-file/) for settings description - look for settings with `filter.`

**Comment by pierrot10** [comment]
Dear vitalidze
Thank for you reply. This I did. I set filter.distance to 3. I tried and I did not record position if I stay in a perimeter of 3meter. It's not what I need, it to supress position which are more than 10km from two point.
For exemple, when the satellite are not available, I got postion with the cell tower. Some time it give a position wich are 10km from my poit and some more fer and sometime in a other country. Then if I look my map, I can see my device which move to an exiatnt position and come back to my normal path. It the reason why I suggest that the latest position is save, it look the two previous position and remove the second if it is a position more far than x kilometer. I think we can not control the only the latest position. Because if I switch off the devise and I turn on while I am in another city, it will not record any position. The goal is to remove incosistent or incoherent position base on kilometer.

**Comment by vitalidze** [comment]
You are correct that `filter.distance` will not work for you as it's purpose is to filter out positions that have distance less than specified one. Can you check `valid` flag of these 'jumped' positions? If it is set to `false` then maybe `filter.invalid` may work for you.
